src/main/java/com/google/api/generator/engine/ast/JavaEscaper.java

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.api.generator.engine.ast;
+
+import com.google.common.escape.Escaper;
+import com.google.common.escape.Escapers;
+
+public class JavaEscaper {
+  public static class SpecialSequenceEscaper extends Escaper {
+    @Override
+    public String escape(String sourceString) {
+      Escaper esc =
+          Escapers.builder()
+              .addEscape('\t', "\\t")
+              .addEscape('\b', "\\b")
+              .addEscape('\n', "\\n")
+              .addEscape('\r', "\\r")
+              .addEscape('\f', "\\f")
+              .addEscape('"', "\\\"")
+              .addEscape('\\', "\\\\")
```

Comment:
sorry, what do you mean by escaping others with leading `"`?
`addEscape(char, string)` helps us to replace escape characters, the following string `"\\r", "\\f", "\\t"` etc only stands for replacement strings, it does not have to start with `"`
Please correct me if I'm not understanding correctly. Thanks!

Reply:
(TIL that  GitHub may eat certain characters like \\)

Retyping: If we're already escaping \\, do we still need to escape the other characters (like \\t) with a leading \\?

Reply:
oh I got your point and yes this is necessary.
when we pass in string value `\` it won't be a single `\` because it's special character. The only way we wanna create string `"\"` is passing in `"\\"`. While `\t` is different because it's a tab, we can directly print a string value as `"\t"` instead of adding one more `\`. 
So as unit test (writeStringObjectValue_escapeCharacter) indicated, for the string value `" \t \b \r \f \\ \n "` the escaper will find `\\` and `\t` separately and replace them, only finding `\\` cannot match `\t`.

Reply:
Sounds good, thanks for clarifying.